qiskit_experiments/tomography/tomography_experiment.py

```diff
@@ -0,0 +1,307 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Quantum Tomography experiment
+"""
+
+import copy
+from typing import Union, Optional, Iterable, List, Tuple
+from itertools import product
+import numpy as np
+from qiskit.circuit import QuantumCircuit, Instruction
+from qiskit.circuit.library import Permutation
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+import qiskit.quantum_info as qi
+from qiskit_experiments.base_experiment import BaseExperiment, Options
+from .basis import BaseTomographyMeasurementBasis, BaseTomographyPreparationBasis
+from .tomography_analysis import TomographyAnalysis
+
+
+class TomographyExperiment(BaseExperiment):
+    """Base experiment for quantum state and process tomography"""
+
+    __analysis_class__ = TomographyAnalysis
+
+    @classmethod
+    def _default_experiment_options(cls):
+        return Options(basis_indices=None)
+
+    def __init__(
+        self,
+        circuit: Union[QuantumCircuit, Instruction, BaseOperator],
+        measurement_basis: Optional[BaseTomographyMeasurementBasis] = None,
+        measurement_qubits: Optional[Iterable[int]] = None,
+        preparation_basis: Optional[BaseTomographyPreparationBasis] = None,
+        preparation_qubits: Optional[Iterable[int]] = None,
+        basis_indices: Optional[Iterable[Tuple[List[int], List[int]]]] = None,
```

Comment:
Seems like this API is bit problematic, because instruction order is hard coded in the basis classes and users need to read the code to map instructions to index (actually those instructions are stored as matrices, so the basis is not quite clear). Can this be string-label based? i.e. in basis class storing instructions with labels and here internally maps the labels to instructions.

Reply:
I find strings quite cumbersome. This way all bases can use same integer indices and so only the number of elements matters when constructing them. Basis class documentation should (but doesn't yet) say what indices correspond to what basis matrix.

Reply:
Can you update args documentation to refer to the basis class docstring to guide users to there?